### PR TITLE
Introduce supplier presenter & simplify manufacturer presenter

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Finalizes the migration to object presenters in front office, following https://github.com/PrestaShop/PrestaShop/pull/31309. It allows to lazy load data and use multiple image formats. Introduces presenter for supplier and some minor simplification on ManufacturerLazyArray introduced previously.
| Type?             | new feature
| Category?         | FO
| BC breaks?        | yes - different template structure
| Deprecations?     | no
| How to test?      | Install this, install https://github.com/PrestaShop/classic-theme/pull/117 and see that everything works the same.
| Fixed ticket?     | 
| Related PRs       | https://github.com/PrestaShop/classic-theme/pull/117, https://github.com/PrestaShop/autoupgrade/pull/614
| Sponsor company   | 

### BC breaks
- Theme change on suppliers page, some template modifications and `$suppliers` instead of `$brands` passed to the template.

### New hooks
- actionPresentSupplier
- actionPresentManufacturer

### CODE REVIEW

**Review by tleon — APPROVED** [review]

File: src/Adapter/Presenter/Supplier/SupplierLazyArray.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    public function __construct(
+        array $supplier,
+        Language $language,
+        ImageRetriever $imageRetriever,
+        Link $link
+    ) {
+        $this->supplier = $supplier;
+        $this->language = $language;
+        $this->imageRetriever = $imageRetriever;
+        $this->link = $link;
```
Comment (M0rgan01) [inline src/Adapter/Presenter/Supplier/SupplierLazyArray.php:66]:
```suggestion
    public function __construct(
        protected array $supplier,
        private Language $language,
        private ImageRetriever $imageRetriever,
        private Link $link
    ) {
```

File: src/Adapter/Presenter/Supplier/SupplierPresenter.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    /**
+     * @var Link
+     */
+    protected $link;
+
+    public function __construct(Link $link)
+    {
+        $this->link = $link;
+        $this->imageRetriever = new ImageRetriever($link);
+    }
```
Comment (M0rgan01) [inline src/Adapter/Presenter/Supplier/SupplierPresenter.php:51]:
```suggestion
    public function __construct(protected Link $link)
    {
        $this->imageRetriever = new ImageRetriever($link);
    }
```

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

Auto tests are green : https://github.com/florine2623/testing_pr/actions/runs/5313125383


https://github.com/PrestaShop/PrestaShop/assets/16019289/61b8bdc7-204a-4554-a918-9dd962e5a4ad

LGTM. It is QA ✅ !

**Comment by jolelievre** [comment]
Thanks @Hlavtox and @florine2623

**Pull request merged**